pkg/cmd/util/clientcmd/errors.go

```diff
@@ -1,27 +1,31 @@
 package clientcmd
 
 import (
+	"fmt"
 	"strings"
 
 	kerrors "k8s.io/kubernetes/pkg/api/errors"
 	"k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
 )
 
 const (
-	unknownReason                     = 0
-	noServerFoundReason               = 1
-	certificateAuthorityUnknownReason = 2
-	configurationInvalidReason        = 3
+	unknownReason = iota
+	noServerFoundReason
+	certificateAuthorityUnknownReason
+	configurationInvalidReason
+	tlsOversizedRecordReason
 
 	certificateAuthorityUnknownMsg = "The server uses a certificate signed by unknown authority. You may need to use the --certificate-authority flag to provide the path to a certificate file for the certificate authority, or --insecure-skip-tls-verify to bypass the certificate check and use insecure connections."
 	notConfiguredMsg               = `The client is not configured. You need to run the login command in order to create a default config for your server and credentials:
   oc login
 You can also run this command again providing the path to a config file directly, either through the --config flag of the KUBECONFIG environment variable.
 `
+	tlsOversizedRecordMsg = `Unable to connect to %[2]s using TLS: %[1]s.
+Ensure the specified server supports HTTPS.`
 )
 
 // GetPrettyMessageFor prettifys the message of the provided error
-func GetPrettyMessageFor(err error) string {
+func GetPrettyMessageFor(err error, a ...interface{}) string {
```

Comment:
-1 on varargs passed to rangom format strings... if you want a server name, pass it